src/data/list/indexes.lean

```diff
@@ -16,6 +18,27 @@ namespace list
 
 variables {α : Type u} {β : Type v}
 
+section map_with_index
+
+lemma map_with_index_core_eq (l : list α) (f : ℕ → α → β) (n : ℕ) :
+  l.map_with_index_core f n = l.map_with_index (λ i a, f (i + n) a) :=
+begin
+  induction l with hd tl hl generalizing f n,
+  { simp [list.map_with_index, list.map_with_index_core] },
+  { rw [list.map_with_index],
+    simp [list.map_with_index_core, hl, add_left_comm, add_assoc, add_comm] }
```

Comment:
Can the `rw` and `simp` be combined into one `simp`?

Reply:
I'm not sure, there is something more to do than to concatenate the list and just use `simp`, if so, it produces a deterministic timeout, is there something more than I can do?

Reply:
Oh okay, I didn't see the `simp` loop. Then the proof is fine as-is. I've pushed a commit to remove the redundant `list` namespace prefixes.